src/puppetlabs/dujour/version_check.clj

```diff
@@ -130,7 +133,7 @@
   ([request-values update-server-url callback-fn]
     (validate-config! request-values update-server-url)
     (future
-      (let [key-map (and [:certname :cacert] (keys request-values))
+      (let [key-map (vec (set/intersection (set [:cacert :certname]) (set (keys request-values))))
```

Comment:
If we're just trying to extract `:cacert` and `certname` from `request-values` if it exists, then this could be simplified to:

```
key-map (vec (keys (select-keys request-values [:cacert :certname])))
```